bugshaker/src/main/java/com/github/stkent/bugshaker/BugShaker.java

```diff
@@ -44,277 +43,241 @@
+		public void onActivityStopped(final Activity activity) {
+			feedbackEmailFlowManager.onActivityStopped();
+		}
+	};
+
+	/**
+	 * @param application the embedding application
+	 * @return the singleton <code>BugShaker</code> instance
+	 */
+	public static BugShaker get(@NonNull final Application application) {
+		synchronized (BugShaker.class) {
+			if (sharedInstance == null) {
+				sharedInstance = new BugShaker(application);
+			}
+		}
+
+		return sharedInstance;
+	}
+
+
+
+	public BugShaker(@NonNull final Application application) {
+		this.application = application;
+//
+//		SharedPreferences settings = application.getApplicationContext()
+//			.getSharedPreferences(EMAIL_ADDRESS, Context.MODE_PRIVATE);
+
+	}
+
+	/**
+	 * (Required) Defines one or more email addresses to send bug reports to. This method MUST be
+	 * called before calling <code>assemble</code>. This method CANNOT be called after calling
+	 * <code>assemble</code> or <code>start</code>.
+	 *
+	 * @param emailAddresses one or more email addresses
+	 * @return the current <code>BugShaker</code> instance (to allow for method chaining)
+	 */
+	public BugShaker setEmailAddresses(@NonNull final String... emailAddresses) {
+		if (assembled || startAttempted) {
+			throw new IllegalStateException(
+				"Configuration must be complete before calling assemble or start");
+		}
+		Set<String> stringSet = new HashSet<String>(Arrays.asList(emailAddresses));
+
+//		this.emailAddresses = emailAddresses;
+		SharedPreferencesUtil.save(application.getApplicationContext(), stringSet);
+		return this;
+	}
+
+	/**
+	 * (Optional) Defines a custom subject line to use for all bug reports. By default, reports will
+	 * use the string defined in <code>DEFAULT_SUBJECT_LINE</code>. This method CANNOT be called
+	 * after calling <code>assemble</code> or <code>start</code>.
+	 *
+	 * @param emailSubjectLine a custom email subject line
+	 * @return the current <code>BugShaker</code> instance (to allow for method chaining)
+	 */
+	public BugShaker setEmailSubjectLine(@NonNull final String emailSubjectLine) {
+		if (assembled || startAttempted) {
+			throw new IllegalStateException(RECONFIGURATION_EXCEPTION_MESSAGE);
+		}
+
+//		this.emailSubjectLine = emailSubjectLine;
+		SharedPreferencesUtil.save(application.getApplicationContext(), emailSubjectLine);
+		return this;
+	}
+
+	/**
+	 * (Optional) Defines a dialog type (native/material) to present when a shake is detected.
+	 * Native dialogs are used by default. This method CANNOT be called after calling
+	 * <code>assemble</code> or <code>start</code>.
+	 *
+	 * @param alertDialogType the dialog type to present
+	 * @return the current <code>BugShaker</code> instance (to allow for method chaining)
+	 */
+	public BugShaker setAlertDialogType(@NonNull final AlertDialogType alertDialogType) {
+		if (assembled || startAttempted) {
+			throw new IllegalStateException(RECONFIGURATION_EXCEPTION_MESSAGE);
+		}
+
+		this.alertDialogType = alertDialogType;
```

Comment:
I think it just ensures that there will only ever be one instance of Bugshaker, because it throws an exception if you have already assembled or attempted to start, but Bugshaker doesn't really need an alertDialogType so we got rid of this.alertDialogType = alertDialogType / its declaration at the beginning of the class / its getter method.